python/pyiceberg/catalog/hive.py

```diff
@@ -0,0 +1,262 @@
+    IntegerType,
+    ListType,
+    LongType,
+    MapType,
+    StringType,
+    StructType,
+    TimestampType,
+    TimeType,
+    UUIDType,
+)
+
+hive_types = {
+    BooleanType: "boolean",
+    IntegerType: "int",
+    LongType: "bigint",
+    FloatType: "float",
+    DoubleType: "double",
+    DateType: "date",
+    TimeType: "string",
+    TimestampType: "timestamp",
+    StringType: "string",
+    UUIDType: "string",
+    BinaryType: "binary",
+    FixedType: "binary",
+    DecimalType: None,
+    StructType: None,
+    ListType: None,
+    MapType: None,
+}
+
+
+class _HiveClient:
+    _transport: TTransport
+    _client: Client
+
+    def __init__(self, url: str):
+        url_parts = urlparse(url)
+        transport = TSocket.TSocket(url_parts.hostname, url_parts.port)
+        self._transport = TTransport.TBufferedTransport(transport)
+        protocol = TBinaryProtocol.TBinaryProtocol(transport)
+
+        self._client = Client(protocol)
+
+    def __enter__(self) -> Client:
+        self._transport.open()
+        return self._client
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._transport.close()
+
+
+class HiveCatalog(Catalog):
+    _client: _HiveClient
+
+    def __init__(self, name: str, properties: Properties, url: str):
+        super().__init__(name, properties)
+        self._client = _HiveClient(url)
+
+    def _storage_descriptor(self, schema: Schema, location: Optional[str]) -> StorageDescriptor:
+        ser_de_info = SerDeInfo(serializationLib="org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe")
+        return StorageDescriptor(
+            self._columns(schema),
+            location,
+            "org.apache.hadoop.mapred.FileInputFormat",
+            "org.apache.hadoop.mapred.FileOutputFormat",
+            serdeInfo=ser_de_info,
+        )
+
+    def _columns(self, schema: Schema) -> List[FieldSchema]:
+        return [FieldSchema(field.name, self._convert_hive_type(field.field_type), field.doc) for field in schema.fields]
+
+    def _convert_hive_type(self, col_type: IcebergType) -> str:
+        if hive_type := hive_types.get(type(col_type)):
+            return hive_type
+        raise NotImplementedError(f"Not yet implemented column type {col_type}")
+
+    def _convert_hive_into_iceberg(self, _: HiveTable) -> Table:
+        # Requires reading the manifest, will implement this in another PR
+        # Also nice to have the REST catalog in first
+        return Table()
+
+    def create_table(
+        self,
+        identifier: Union[str, Identifier],
+        schema: Schema,
+        location: Optional[str] = None,
+        partition_spec: Optional[PartitionSpec] = None,
+        properties: Optional[Properties] = None,
+    ) -> Table:
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        current_time_millis = int(time())
+        tbl = HiveTable(
+            dbName=database_name,
+            tableName=table_name,
+            owner=getuser(),
```

Comment:
Can we refactor this to be a `user` catalog arg that defaults to `getuser`? I know this is what the legacy client did, but I don't think that was a good idea.

Reply:
I also found this a bit misleading. We don't have any notion of an owner in the REST catalog. We could also set it to `pyiceberg`?

Reply:
I think this is minor but we should avoid breaking behavior. I'd continue to use `getuser` here, but not in the REST catalog.

Reply:
I had to mock the `getuser` since it also doesn't play nice with the tests when you do `assert_called_with` (differnt user in the CI)

Reply:
@Fokko, instead of mocking it, why not add an option to the catalog config properties like I suggested? Then you could just set it in config rather than mocking `getuser`.

Reply:
Ah, I didn't get the part about getting it from the properties. I like that!